jdaviz/core/helpers.py

```diff
@@ -35,3 +39,197 @@ def __init__(self, app=None):
 
     def load_data(self, data, parser_reference=None, **kwargs):
         self.app.load_data(data, parser_reference=parser_reference, **kwargs)
+
+    @property
+    def fitted_models(self):
+        """
+        Returns the fitted model parameters.
+
+        Returns
+        -------
+        parameters : dict
+            dict of Quantity arrays, or None.
+        """
+
+        return self.app.fitted_models
+
+    def get_models(self, models=None, model_label="default", x=None, y=None):
+        """
+        Loop through all models and output models of the label model_label.
+        If x or y is set, return model_labels of those (x, y) coordinates.
+        If x and y are None, print all models regardless of coordinates.
+
+        Parameters
+        ----------
+        models : dict
+            A dict of models, with the key being the label name and the value
+            being an `astropy.modeling.CompoundModel` object. Defaults to
+            `self.fitted_models` if no parameter is provided.
+        model_label : str
+            The name of the model that will be found and returned. If it
+            equals default, every model present will be returned.
+        x : int
+            The x coordinate of the model spaxels that will be returned.
+        y : int
+            The y coordinate of the model spaxels that will be returned.
+
+        Returns
+        -------
+        :dict: dictionary of the selected models.
+        """
+        selected_models = {}
+        # If models is not provided, use the app's fitted models
+        if not models:
+            models = self.fitted_models
+
+        # Loop through all keys in the dict models
+        for label in models:
+            # If no label was provided, use label name without coordinates.
+            if model_label == "default" and " (" in label:
+                find_label = label.split(" (")[0]
+            # If coordinates are not present, just use the label.
+            elif model_label == "default":
+                find_label = label
+            else:
+                find_label = model_label
+
+            # If x and y is set, return keys that match the model plus that
+            # coordinate pair. If only x or y is set, return keys that fit
+            # that value for the appropriate coordinate. If neither is set,
+            # add the model object to the selected_models dict with the label
+            # as the key.
+            if x and y:
+                find_label = f"{find_label} ({x}, {y})"
+            elif x:
+                find_label = r"{} \({}, .+\)".format(find_label, x)
+            elif y:
+                find_label = f"{find_label} (.+, {y})"
```

Comment:
Here too, although specifying y was working for me. Maybe better to be consistent anyway.
```suggestion
                find_label = f"{find_label} \(.+, {y}\)"
```